homeassistant/components/mqtt/entity.py

```diff
@@ -1472,7 +1467,7 @@ def _message_callback(
             _LOGGER.warning(exc)
             return
 
-        if attributes is not None and self._attrs_have_changed(attrs_snapshot):
```

Comment:
It seems to me more cleanup is needed here if the helper is removed.
It seems not right we are tracking attributes, but not using it.
I am not sure if this might break users automation, but as ha core will filter out the duplicates, this should not be an issue,